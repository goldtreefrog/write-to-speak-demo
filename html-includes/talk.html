<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="keywords" content="">
  <meta name="description" content="">
  <meta name="robots" content="">
  <meta name="author" content="Margaret Blauvelt">
  <title>Write to Speak</title>
  <link rel="stylesheet" href="index.css">
</head>

<body>
  <main id="write" role="main">
    <section id="snippets">
      <form action="#">
        <h2>Talk</h2>
        <h3>Click on a snippet to make it speak</h3> @@include('./partials/_snippets.html')
      </form>
    </section>
  </main>

</body>

</html>