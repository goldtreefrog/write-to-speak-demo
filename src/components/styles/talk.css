p.page-instructions {
  text-align: left;
  margin: 16px 2% 4px;
  clear: both;
}

.page-snippets {
  display: block;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  clear: both;
  overflow: hidden;
}

.snippet-talk {
  border-collapse: collapse;
  width: 98%;
  background-color: #cdf;
  border-radius: 5px;
  border: solid 2px #111;
  margin-top: 10px;
  margin-bottom: -25px;
  z-index: 0;
  position: relative;
}
.snippet-talk div {
  z-index: 0;
  position: relative;
}
.seb-container {
  width: 100%;
}
.snippet-expand-button {
  z-index: 1;
  padding: 0;
  border: none;
  background: none;
  font-size: 1rem;
  position: relative;
  margin-right: 6px;
  margin-left: auto;
  display: block;
}
.snippet-expand-button:focus,
.snippet-expand-button:hover {
  background: none;
  background-color: white;
  color: #111;
}
div.snippet-expanded div {
  overflow: visible;
  z-index: 1;
  background-color: transparent;
  display: block;
  position: absolute;
  top: 6px;
  left: 6px;
}
div.snippet-expanded {
  height: 80vh;
  overflow-y: auto;
  z-index: 1;
  background-color: #eee;
  position: fixed;
  top: 13vh;
  left: 1%;
  width: 97.5%;
}
button.snippet-expanded {
  position: fixed;
  top: 87%;
  right: 5px;
  font-size: 150%;
  background-color: #aaa;
}

.snippet-faded {
  visibility: hidden;
}
@media only screen and (min-width: 183px) {
  div.snippet-expanded {
    height: 84vh;
    position: fixed;
    top: 10vh;
  }
  button.snippet-expanded {
    top: 88%;
    right: 11px;
    font-size: 150%;
  }
}
@media only screen and (min-width: 370px) {
  div.snippet-expanded {
    height: 80vh;
    position: fixed;
    top: 15vh;
  }
  button.snippet-expanded {
    top: 89%;
    right: 17px;
  }
}
/* Tablet or small screen, not phone, 2 snippets across */
@media only screen and (min-width: 440px) {
  section {
    padding-left: 2%;
    padding-right: 2%;
  }
  .page-snippets {
    width: 100%;
    display: block;
  }
  .snippet-talk-line {
    margin: 6px 1%;
    width: 48%;
    height: 3.5em;
    float: left;
  }
  .snippet-talk {
    height: 3.5em;
  }
  .snippet-talk div {
    height: 2.6em;
  }

  div.snippet-expanded {
    width: 96%;
    top: 16vh;
    height: 78vh;
  }
  button.snippet-expanded {
    top: 87%;
    right: 23px;
  }
}
@media only screen and (min-width: 540px) {
  section {
    padding-left: 3%;
    padding-right: 3%;
  }
  .page-snippets {
    margin: 0 0.9%;
  }
  .snippet-talk-line {
    width: 31%;
    float: left;
  }

  div.snippet-expanded {
    width: 80%;
    margin-left: 10%;
    margin-right: 10%;
    top: 10vh;
    height: 84vh;
    display: block;
  }
  button.snippet-expanded {
    top: 88%;
    right: 10%;
  }
}

/* Non-mobile, snippets  4 snippets across */
@media only screen and (min-width: 640px) {
  section {
    padding-left: 4%;
    padding-right: 4%;
  }
  .page-snippets {
    margin: 0 1%;
  }
  .snippet-talk-line {
    width: 22%;
    float: left;
  }
}
