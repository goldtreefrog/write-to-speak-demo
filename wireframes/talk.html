<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="keywords" content="">
  <meta name="description" content="">
  <meta name="robots" content="">
  <meta name="author" content="Margaret Blauvelt">
  <title>Write to Speak</title>
  <link rel="stylesheet" href="index.css">
</head>

<body>
  <main id="write" role="main">
    <section id="snippets">
      <form action="#">
        <h2>Talk</h2>
        <h3>Click on a snippet to make it speak</h3> <button type="button" name="snip01" id="snip01" value="Mom">Mom</button>
<button type="button" name="snip02" id="snip02" value="Dad">Dad</button>
<button type="button" name="snip03" id="snip03" value="I am hungry">I am hungry</button>
<button type="button" name="snip04" id="snip04" value="I love you">I love you</button>
<button type="button" name="snip05" id="snip05" value="Once upon a time, when it was a dark and stormy night, there was a terrible...">Once upon a time, when it was a dark and stormy night, there was a terrible...</button>
<button type="button" name="snip06" id="snip06" value="Joe is my best friend. I know you will like him.">Joe is my best friend. I know you will like him.</button>
      </form>
    </section>
  </main>

</body>

</html>