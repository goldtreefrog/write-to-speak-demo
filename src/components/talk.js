import React from "react";
// import Snippets from "./snippets";
import Snippet from "./snippet";
import "./styles/talk.css";
import { snippetData as snippets } from "./../tests/fixtures/snippet-data.js";

class Talk extends React.Component {
  render() {
    return (
      <section id="snippets">
        <h2>Talk</h2>
        <p>Click on a snippet to make it speak</p>
        {snippets.map(snippet => <Snippet text={snippet.text} id={snippet.id} orderkey={snippet.orderkey} key={snippet.orderkey} />)}
        {/* <Snippets snippets={snippets} /> */}
      </section>
    );
  }
}

export default Talk;
