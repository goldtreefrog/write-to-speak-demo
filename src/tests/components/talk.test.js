import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { Talk } from "./../../components/talk.js";
import Snippet from "./../../components/snippet.js";
import { snippets } from "./../fixtures/snippet-data.js";
import { feedback } from "./../fixtures/feedback-data.js";

describe("<Talk />", () => {
  it("renders Talk page and matches snapshot", () => {
    const wrapper = shallow(
      <Talk snippets={snippets} feedback={feedback} loggedIn={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders Talk page when logged in", () => {
    const wrapper = shallow(
      <Talk snippets={snippets} feedback={feedback} loggedIn={true} />
    );
    expect(wrapper.contains(<h2>Talk</h2>)).toEqual(true);
    expect(
      wrapper.contains(
        <p className="page-instructions">Click on a snippet to hear it:</p>
      )
    ).toEqual(true);
  });

  it("Redirects to Login page when not logged in", () => {
    const wrapper = shallow(
      <Talk snippets={snippets} feedback={feedback} loggedIn={false} />
    );
    expect(wrapper.contains(<h2>Talk</h2>)).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });

  it("Can expand a snippet", () => {
    const wrapper = shallow(
      <Talk snippets={snippets} feedback={feedback} loggedIn={true} />
    );
    wrapper.setState({
      expandedSnippet: false,
      expandedSnippetId: ""
    });
    wrapper.instance().expandSnippet("123");
    expect(wrapper.state("expandedSnippet")).toEqual(true);
    expect(wrapper.state("expandedSnippetId")).toEqual("123");
  });
});
