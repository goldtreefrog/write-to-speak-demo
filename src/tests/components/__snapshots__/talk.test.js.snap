// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<section
  id="snippets"
>
  <Snippet
    id={123}
    key="0"
    orderkey={0}
    text="Hi there"
  />
  <Snippet
    id={312}
    key="3"
    orderkey={3}
    text="Mom"
  />
  <Snippet
    id={666}
    key="2"
    orderkey={2}
    text="Dad"
  />
  <Snippet
    id={427}
    key="1"
    orderkey={1}
    text="come talk with me"
  />
  <Snippet
    id={555}
    key="6"
    orderkey={6}
    text="Once upon a time, when it was a dark and stormy night, Sandy heard a tremendously loud 'boom!'"
  />
</section>
`;
